x/inflation/types/v1beta2/inflation_calculator.go

```diff
@@ -0,0 +1,68 @@
+package v1beta2
+
+import (
+	"math"
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+var (
+	// these are set during NewApp and used in InflationCalculator
+
+	GenesisTime            time.Time
+	InflationParamSubspace paramstypes.Subspace
+)
+
+func InflationCalculator(ctx sdk.Context, minter minttypes.Minter, params minttypes.Params, bondedRatio sdk.Dec) sdk.Dec {
+	var inflationParams Params
+	InflationParamSubspace.GetParamSet(ctx, &inflationParams)
+	tHalf := float64(inflationParams.InflationDecayFactor)
+	initialInflation, err := strconv.ParseFloat(inflationParams.InitialInflation, 32)
+	if err != nil {
+		panic(err)
+	}
+
+	// Number of hours in an year = 8766 (averaging the leap year hours)
+	// Number of minutes in an hour = 60
+	// Number of seconds in a minute = 60
+	// => 60 * 60 * 8766 = Number of seconds per year
+	t := ctx.BlockTime().Sub(GenesisTime).Seconds() / (60 * 60 * 8766) // years passed
```

Comment:
is `ctx.BlockTime()` the same across each node @boz  ? I haven't spent much time thinking about it. I mostly try and think in block height.

Reply:
Yes, I believe it's equivalent.  I can double check.

I still think that the above calculation should not use integer division - it will be `0` for a whole year, then `1` for a year, then `2`, etc..., causing unnecessary and unwanted discontinuities on the anniversary of genesis.

Reply:
It is not using integer division.

`ctx.BlockTime().Sub(GenesisTime).Seconds()` returns a `float64`, so it is indeed a continuous function, and not a step function. i.e., it can have values like 0.1 -> 0.2 -> 0.3 -> and so on...

Reply:
I think `ctx.BlockTime().Sub(GenesisTime)` returns a `time.Duration` type. So can we just call `.Milliseconds()` to get an "exact" numerical value here? Millisecond level precision should be more than accurate enough for calculating interest I would think